### PR TITLE
Add dual systemd / upstart support

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Tested locally. Works as expected. (Aside from local permissions issues.)

File: pkg/dmqnode-common.pkg [hunk 1]
```diff
@@ -14,5 +14,6 @@ OPTS.update(
 # Replace existing args without rebinding
 ARGS[:] = [
         "README.rst=/usr/share/doc/" + VAR.shortname + "/README.rst",
-        "deploy/logrotate/dmqnode-logs=/etc/logrotate.d/"
+        "deploy/logrotate/dmqnode-logs=/etc/logrotate.d/",
+        "./deploy/systemd/dmq@.service=/lib/systemd/system/dmq@.service"
```
Comment (nemanja-boric-sociomantic) [inline pkg/dmqnode-common.pkg]: Maybe just make it `deploy/systemd/dmq@.service` to be consistent with the other lines.

File: deploy/systemd/dmq@.service [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# of DMQ node if the server is about to die.
+OOMScoreAdjust=-950
+
+# Note infinity is used instead of unlimited
+LimitNOFILE=100000:100000
+LimitMEMLOCK=infinity:infinity
+LimitCORE=infinity:infinity
+
+[Install]
+WantedBy = multi-user.target
```
Comment (nemanja-boric-sociomantic) [inline deploy/systemd/dmq@.service]: Missing newline.

**Comment by gavin-norman-sociomantic** [comment]
Updated.

File: deploy/systemd/dmq@.service [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+Group=core
+
+# Could be even 1000 (to disable oom killing),
+# but set to -950 for the OOM killer to be able to kill an instance
+# of DMQ node if the server is about to die.
+OOMScoreAdjust=-950
+
+# Note infinity is used instead of unlimited
+LimitNOFILE=100000:100000
+LimitMEMLOCK=infinity:infinity
```
Comment (nemanja-boric-sociomantic) [inline deploy/systemd/dmq@.service]: Oh, you don't need this! I'll adjust the guide.

Comment (nemanja-boric-sociomantic) [reply deploy/systemd/dmq@.service]: (it's relevant to dhtnode, not for other apps)

Comment (gavin-norman-sociomantic) [reply deploy/systemd/dmq@.service]: Ha yes. Too much copypasta! :D

Comment (gavin-norman-sociomantic) [reply deploy/systemd/dmq@.service]: Updated.

**Pull request merged**